pkg/providers/v1/aws.go

```diff
@@ -1963,6 +1982,42 @@ func (c *Cloud) buildELBSecurityGroupList(ctx context.Context, serviceName types
 	return sgList, setupSg, nil
 }
 
+// createSecurityGroupRulesForNLB creates and configures security group rules for a Network Load Balancer (NLB).
+//
+// This function sets up ingress and egress rules for the specified security group (sgID) based on the provided
+// port mappings and source IP ranges.
+//
+// Parameters:
+//   - sgID: The ID of the security group to configure.
+//   - ports: A slice of nlbPortMapping objects that define the port mappings for the NLB.
+//   - ec2SourceRanges: A slice of *ec2.IpRange objects specifying the source IP ranges for the rules.
+//
+// Returns:
+//   - error: An error if any issue occurs while creating or applying the security group rules.
+func (c *Cloud) createSecurityGroupRulesForNLB(ctx context.Context, sgID string, ports []nlbPortMapping, ec2SourceRanges []ec2types.IpRange) error {
+	ingressRules := NewIPPermissionSet()
+
+	// Question: Do we need to manage outbound rules? Egress rules are not managed in v1 (CLB),
+	// currently the logic is duplicating v1.
```

Comment:
What would be the good initial approach here? The CLB uses `0/0`(allow all) as outbound rules - default rule when creating an SG. Should we need to keep it or enhance that behavior on NLB-SG's?
TODO check how SGs provisioned in ALBC behave;